android-components/components/feature/search/src/main/java/mozilla/components/feature/search/internal/SearchUrlBuilder.kt

```diff
@@ -44,15 +46,21 @@ internal class SearchUrlBuilder(
 
     private fun buildUrl(template: String, searchTerms: String): String {
         val templateUri = Uri.decode(template)
-        val urlWithSubstitutions = paramSubstitution(templateUri, Uri.encode(searchTerms))
+        val inputEncoding = searchEngine.inputEncoding ?: "UTF-8"
```

Comment:
nit: instead of `inputEncoding` being nullable could we just default to "UTF-8"?

Reply:
default is [UTF-8](https://github.com/dewitt/opensearch/blob/master/opensearch-1-1-draft-6.md#the-inputencoding-element) as OpenSearch format. Although I check old code for Mozilla's search plugins due to no document now, it is default charset (Android is UTF-8) if nothing.